### PR TITLE
Fixed #32720 -- Added configuration and docs for Sphinx link checker.

### PR DESCRIPTION
**Pull request by pope1ni**
ticket-32720

I've used the [linkcheck builder](https://www.sphinx-doc.org/en/master/usage/configuration.html#options-for-the-linkcheck-builder) in Sphinx to identify a number of broken links that I've fixed, and updated others to use the canonical link so that unnecessary redirects are avoided.

Ideally, the output generated in `docs/_build/linkcheck/output.{json,txt}` would only contain "working" links, but there are some links that are better left redirecting, e.g. for documentation versions such as `https://docs.djangoproject.com/en/stable/` → `https://docs.djangoproject.com/en/3.2/`. There is an open issue for adding a configuration setting to allow certain redirects to be treated as "normal". See https://github.com/sphinx-doc/sphinx/issues/6525#issuecomment-828730361.

Questions:

- [x] Should we add something to `docs/internals/contributing/writing-documentation.txt` about using the linkcheck builder?
- [x] Should we add something to `docs/internals/howto-release-django.txt` about checking before preparing a release?
- [x] ~~If sphinx-doc/sphinx#6525 is resolved, would this check be useful to add to a scheduled action in GitHub?~~
  - *(Opened ticket-32723 to address this at a later date.)*

### CODE REVIEW

**Review by felixxm — COMMENTED** [review]
@pope1ni Thanks for this patch :+1: I reorganized commits, corrected hashes in the security archive, updated links to the mailing lists, and pushed small edits.

File: docs/releases/security.txt [hunk 1]
```diff
@@ -1156,9 +1156,9 @@ XSS via admin login redirect. `Full description
 Versions affected
 ~~~~~~~~~~~~~~~~~
 
-* Django 0.91 :commit:`(patch) <50ce7fb57d>`
-* Django 0.95 :commit:`(patch) <50ce7fb57d>`
-* Django 0.96 :commit:`(patch) <7791e5c050>`
+* Django 0.91 :commit:`(patch) <50ce7fb57d79e8940ccf6e2781f2f01df029b5c5>`
```
[the remark this answers is not in the document]
Comment (pope1ni) [reply docs/releases/security.txt]: Ah, yes. I noticed these and meant to look into it. Thanks!

File: docs/conf.py [hunk 2]
```diff
@@ -18,7 +18,7 @@
 #
 # Python's default allowed recursion depth is 1000 but this isn't enough for
 # building docs/ref/settings.txt sometimes.
-# https://groups.google.com/d/topic/sphinx-dev/MtRf64eGtv4/discussion
+# https://groups.google.com/forum/#!topic/sphinx-dev/MtRf64eGtv4/discussion
```
[the remark this answers is not in the document]
Comment (pope1ni) [reply docs/conf.py]: Ah, these `…/g/…` links are much better 💯

**Review by pope1ni — COMMENTED** [review]
Thanks for the tweaks Mariusz.

File: docs/internals/howto-release-django.txt [hunk 3]
```diff
@@ -104,8 +104,8 @@ any time leading up to the actual release:
    uncommitted changes for the release.
 
 #. Proofread the release notes, including looking at the online
-   version to catch any broken links or reST errors, and make sure the
-   release notes contain the correct date.
+   version to :ref:`catch any broken links <documentation-link-check>` or reST
+   errors, and make sure the release notes contain the correct date.
```
[the remark this answers is not in the document]
Comment (felixxm) [reply docs/internals/howto-release-django.txt]: Updated.

**Pull request merged**